### ISSUE TITLE
Breaks when asked to track the empty string

### ISSUE THREAD

**Issue by snan** [GothenburgBitFactory/timewarrior#388]
When given `timew start ''` it starts an interval that it can't stop. It complains that you cannot overlap intervals. Editing the data file sorts things out.

I do not want to track empty tags; one suggested fix is for `timew start` to throw away empty tags, reporting an error message while doing so, and start tracking as if there were no tag. Another route is for `timew stop` to be able to handle files in the weird state that `timew start ''` leaves them in.

**Comment by sruffell** [comment]
My vote would be to print an error message if there are any empty tags.

That said, you may be interested in knowing that since 1.3.0 (specifically commit c1f3d8be72e2f175477ba) timewarrior would not get into a broken state in this case since it would fail the test that the new database entry can be parsed into the same value.

For example:
```
$ src/timew start foo 10min ago
Note: 'foo' is a new tag.
Tracking foo
  Started 2020-09-24T05:06:42
  Current               16:42
  Total               0:10:00
$ src/timew start ''
Recorded foo
  Started 2020-09-24T05:06:42
  Ended                 16:46
  Total               0:10:04
Note: '' is a new tag.
Internal error. Failed encode / decode check.
$ src/timew stop
Recorded foo
  Started 2020-09-24T05:06:42
  Ended                 16:52
  Total               0:10:10
$ src/timew summary

Wk  Date       Day Tags   Start     End    Time   Total
W39 2020-09-24 Thu foo  5:06:42 5:16:52 0:10:10 0:10:10

                                                0:10:10

```

**Comment by snan** [comment]
In that case, this isn't an issue anymore. Thank you for having fixed this in the past. I was on an older version.

**Comment by snan** [comment]
In case people were curious, the reason I got into trouble in the first place was that I was running from cron:

```
timew-dwim () {
    deduce-tag|empty-stdin && timew stop || timew start "$(deduce-tag)"
}
```
The intent is that when deduce-tag doesn't output anything, timew will stop tracking, but when it does output text, timew will track that as a tag.  
Buuuut since timew stop returns non-zero when there's no tracking (not sure that's a good idea, I mean, is it really an error that it didn't need to stop because tracking was already stopped?) I would get both branches and thus start tracking an empty tag. I fixed it right away with a

```
timew-stop-return () {
    timew stop
    return 0
}
```
wrapper.

Thank you for making this wonderful app.♥
